hudi-spark/src/main/java/org/apache/hudi/DataSourceUtils.java

```diff
@@ -77,6 +80,11 @@ public static Object getNestedFieldVal(GenericRecord record, String fieldName, b
 
       // return, if last part of name
       if (i == parts.length - 1) {
+
+        if (isLogicalTypeDate(valueNode, part)) {
+          return LocalDate.ofEpochDay(Long.parseLong(val.toString()));
```

Comment:
May not be related to this PR. but is time zone information lost when we convert from Date to Long (row -> avro)?

Reply:
From my understanding, I don't think Spark's `DateType` has a timezone. It should be using the Local TimeZone. Same goes for `LocalDate`.